src/tests/storage/clipperData_tests.ts

```diff
@@ -0,0 +1,121 @@
+
+import {ClipperData} from "../../scripts/storage/clipperData";
+import {ClipperStorageKeys} from "../../scripts/storage/clipperStorageKeys";
+
+import {MockStorage} from "./mockStorage";
+
+let mockStorage: MockStorage;
+
+QUnit.module("clipperData", {
+	beforeEach: () => {
+		mockStorage = new MockStorage();
+	}
+});
+
+test("getAndCacheFreshValue should not store anything if it's retrieving notebooks and no userInformation is in storage", (assert: QUnitAssert) => {
+	let done = assert.async();
+
+	let key = ClipperStorageKeys.cachedNotebooks;
+	let parsedResponse = {};
+	let expectedTimeStampedData = {
+		parsedResponse: JSON.stringify(parsedResponse),
+		request: undefined
+	};
+	let getRemoteValue = () => {
+		return Promise.resolve(expectedTimeStampedData);
+	};
+
+	let clipperData = new ClipperData(mockStorage);
+	clipperData.getAndCacheFreshValue(key, getRemoteValue, 0).then((timeStampedData) => {
+		strictEqual(mockStorage.getValue(key), undefined,
+			"Notebooks should not be cached if userInformation does not exist in storage");
+	}, (error) => {
+		ok(false, "reject should not be called");
+	}).then(() => {
+		done();
+	});
+});
+
+test("getAndCacheFreshValue should store notebooks if it's retrieving notebooks and userInformation is in storage", (assert: QUnitAssert) => {
+	let done = assert.async();
+
+	let key = ClipperStorageKeys.cachedNotebooks;
+	let parsedResponse = {};
+	let expectedTimeStampedData = {
+		parsedResponse: JSON.stringify(parsedResponse),
+		request: undefined
+	};
+	let getRemoteValue = () => {
+		return Promise.resolve(expectedTimeStampedData);
+	};
+
+	mockStorage.setValue(ClipperStorageKeys.userInformation, "{ name: Leeeeeroy }");
+
+	let clipperData = new ClipperData(mockStorage);
+	clipperData.getAndCacheFreshValue(key, getRemoteValue, 0).then((timeStampedData) => {
+		let actualStored: TimeStampedData = JSON.parse(mockStorage.getValue(key));
+		deepEqual(actualStored.data, {},
+			"Notebooks should be cached if userInformation exists in storage");
+	}, (error) => {
+		ok(false, "reject should not be called");
+	}).then(() => {
+		done();
+	});
+});
+
+test("getAndCacheFreshValue should not store anything if it's setting notebooks and no userInformation is in storage", () => {
+	let key = ClipperStorageKeys.cachedNotebooks;
+	let expectedTimeStampedData = JSON.stringify({
+		parsedResponse: "{ notebooks: {} }",
+		request: undefined
+	});
+
+	let clipperData = new ClipperData(mockStorage);
+	clipperData.setValue(key, expectedTimeStampedData);
+	strictEqual(mockStorage.getValue(key), undefined,
+		"Notebooks should not be stored if userInformation does not exist in storage");
+});
+
+test("getAndCacheFreshValue should store notebooks if it's setting notebooks and userInformation is in storage", () => {
+	let key = ClipperStorageKeys.cachedNotebooks;
+	let expectedTimeStampedData = JSON.stringify({
+		parsedResponse: "{ notebooks: {} }",
+		request: undefined
+	});
+
```

Comment:
Adding my 2c, the reason why I tested this the way I did was because I wanted to do so in a blackbox manner (i.e., we don't know there's a gating strategy as part of the implementation). If we use DI, more tests will need to be added for better coverage.